sys/arduino/SPI.cpp

```diff
@@ -67,25 +67,79 @@ SPIClass::SPIClass(spi_t spi_dev)
     /* Check if default SPI interface is valid */
     BUILD_BUG_ON(ARDUINO_SPI_INTERFACE >= SPI_NUMOF);
     this->spi_dev = spi_dev;
+    this->settings = SPISettings();
+    this->is_transaction = false;
+    rmutex_init(&this->mut);
 }
 
 void SPIClass::beginTransaction(SPISettings settings)
 {
+    rmutex_lock(&mut);
```

Comment:
Although it is not a good idea to start any SPI transfer in an ISR, would it make sens to add an `assert(!irq_is_in())` before locking the mutex?

Reply:
This wouldn't solve the issue, as `spi_acquire()` internally also locks a mutex.

Reply:
> This wouldn't solve the issue, as spi_acquire() internally also locks a mutex.

I know, it's a general problem. @kaspar030 would it make sense to add `assert(!irq_is_in())` in `*mutex` functions that must not be called in ISRs. It would help the developer to recognize the existence of such a problem without any extra byte with `NDEBUG` defined.